### ISSUE TITLE
MultiButton - ClickUp Debounce not handled?

### ISSUE THREAD

**Issue by sergeyken** [poelstra/arduino-multi-button#10]
Hello,

After looking into the source of MultiButton.h I'm under impression, that debouncing is done only for pressing button, but not for its releasing, isn't it?
At first moment the button is !pressed, after _state==StatePressed there is a change: _state=StateClickUp; _new=true;
The call to isReleased() returns (_new && _state == StateClickUp) which is true at this moment.
No check for DEBOUNCE_DELAY is performed at this situation.

Please, correct me if I am wrong?

**Comment by poelstra** [comment]
@sergeyken You're correct. When a switch is released, it immediately indicates `isReleased()`.

Note that any bounces that follow (during the 'turn off') _are_ debounced, but they will be debounced as being part of a 'potential turn on' (which will subsequently be 'dismissed' as being a glitch).

'Graphical' example:
```
__#_#_########_#_#_#__________#_#_########
  | |   |     || | |          | |   |
  | |   |     || | |          | |   +-> isClick B
  | |   |     || | |          +-+-> discarded as glitches
  | |   |     |+-+-> discarded as 'turn on glitches' too
  | |   |     +-> isReleased A
  | |   +-> isClick A
  +-+-> discarded as glitches
```

This behavior leads to faster response times when using 'normal' mechanical switches, which was my original use-case and why I implemented it like this.

For the use-case where turning off might in itself be glitchy, i.e. where a short 'off glitch' should not be treated as turning off, I agree that the current implementation is not ideal.

In practice, I've also used the library with a capacitive touch system (which is a bit glitchy like that) and it worked just fine for me, but YMMV.

**Comment by sergeyken** [comment]
Sorry, there was an unexpected change in my comment. I tried to show the picture like this one:
___#########_##########_________

**Comment by sergeyken** [comment]
OK, thank you for this explanation.

Maybe, this happens rarely, but the situation like this one:
.__########__########_________
would be handled as two separate clicks for those who checked only isClick(), and/or isReleased().
Also isDoubleClick() is detected in this situation, rather than isLongClick() as one could expect.

**Comment by sergeyken** [comment]
OK, thank you for this explanation.

Maybe, this happens rarely, but the situation like this one:
........########..########............
would be handled as two separate clicks for those who checked only isClick(), and/or isReleased().
Also isDoubleClick() is detected in this situation, rather than isLongClick() as one could expect.
